remote/remote_state_s3.go

```diff
@@ -79,22 +79,29 @@ var terragruntOnlyConfigs = []string{
 	"bucket_sse_kms_key_id",
 }
 
+type RemoteStateConfigS3AssumeRole struct {
+	RoleArn     string `mapstructure:"role_arn"`
+	ExternalID  string `mapstructure:"external_id"`
+	SessionName string `mapstructure:"session_name"`
+}
+
 // A representation of the configuration options available for S3 remote state
 type RemoteStateConfigS3 struct {
-	Encrypt          bool   `mapstructure:"encrypt"`
-	Bucket           string `mapstructure:"bucket"`
-	Key              string `mapstructure:"key"`
-	Region           string `mapstructure:"region"`
-	Endpoint         string `mapstructure:"endpoint"`
-	DynamoDBEndpoint string `mapstructure:"dynamodb_endpoint"`
-	Profile          string `mapstructure:"profile"`
-	RoleArn          string `mapstructure:"role_arn"`
-	ExternalID       string `mapstructure:"external_id"`
-	SessionName      string `mapstructure:"session_name"`
-	LockTable        string `mapstructure:"lock_table"` // Deprecated in Terraform version 0.13 or newer.
-	DynamoDBTable    string `mapstructure:"dynamodb_table"`
-	CredsFilename    string `mapstructure:"shared_credentials_file"`
-	S3ForcePathStyle bool   `mapstructure:"force_path_style"`
+	Encrypt          bool                          `mapstructure:"encrypt"`
+	Bucket           string                        `mapstructure:"bucket"`
+	Key              string                        `mapstructure:"key"`
+	Region           string                        `mapstructure:"region"`
+	Endpoint         string                        `mapstructure:"endpoint"`
+	DynamoDBEndpoint string                        `mapstructure:"dynamodb_endpoint"`
+	Profile          string                        `mapstructure:"profile"`
+	RoleArn          string                        `mapstructure:"role_arn"`     // Deprecated in Terraform version 1.6 or newer.
+	ExternalID       string                        `mapstructure:"external_id"`  // Deprecated in Terraform version 1.6 or newer.
+	SessionName      string                        `mapstructure:"session_name"` // Deprecated in Terraform version 1.6 or newer.
+	LockTable        string                        `mapstructure:"lock_table"`   // Deprecated in Terraform version 0.13 or newer.
+	DynamoDBTable    string                        `mapstructure:"dynamodb_table"`
+	CredsFilename    string                        `mapstructure:"shared_credentials_file"`
+	S3ForcePathStyle bool                          `mapstructure:"force_path_style"`
+	AssumeRole       RemoteStateConfigS3AssumeRole `mapstructure:"assume_role"`
```

Comment:
Will be helpful to document/add examples how to define `AssumeRole` in https://github.com/gruntwork-io/terragrunt/blob/master/docs/_docs/04_reference/config-blocks-and-attributes.md

Reply:
done